library/Rules/Email.php

```diff
@@ -5,6 +5,14 @@ class Email extends AbstractRule
 {
     public function validate($input)
     {
-        return is_string($input) && filter_var($input, FILTER_VALIDATE_EMAIL);
+        if (class_exists('\Egulias\EmailValidator\EmailValidator')) {
+            $strictValidator = new \Egulias\EmailValidator\EmailValidator();
+
+            $validationResult = $strictValidator->isValid($input);
+        } else {
+            $validationResult = is_string($input) && filter_var($input, FILTER_VALIDATE_EMAIL);
+        }
```

Comment:
The else part is for using `filter_var` as fall back. If I avoid the else part, the lib should be a hard dependency and not a soft one. I understood you wanted to be soft dep., right?